feature/text_based.py

```diff
@@ -0,0 +1,703 @@
+        else:
+            raise NotImplementedError(f"{optimization_method} has not been implemented.")
+
+        # Sort selection
+        selected.sort()
+
+        return selected
+
+    def _process_data(self, input_df: pd.DataFrame, categories: pd.DataFrame, selection_size: int,
+                      optimization_method: str, cost_metric: str) -> NoReturn:
+        """_process_data
+        We have implemented two versions of this function. In the version here, the categories parameter
+        actually refers to the labels (rather than categories) to be covered. It generates features for input data
+        using a specified featurization method, and creates a matrix where each row corresponds to a label and each
+        column corresponds to a feature.
+        The version of code for generating labels from categories has been moved to utils.py.
+
+        Parameters
+        ----------
+        input_df: pd.DataFrame
+            Input data frame with categories and features of content to select from.
+
+        categories: pd.DataFrame
+            Data frame that contains categories/labels to be covered.
+
+        selection_size: int
+            Number of feature to select.
+
+        optimization_method: str
+            Code prevents performing featurization for the case mentioned in 'option_map' dictionary with
+            random, greedy and exact optimization.
+
+        cost_metric: str
+            Code prevents performing featurization for the case mentioned in 'option_map' dictionary with
+            unicost or diverse cost metric.
+
+        Returns
+        -------
+        No return
+       --------
+        """
+        self.matrix = categories.to_numpy()
+        self._num_rows, self._num_cols = self.matrix.shape
+
+        # option_map contains the tuple of optimization_method, selection_size, and cost_metric
+        # that do not need featurization method
+        option_map = {
+            ("random", True, "diverse"): lambda: None,
+            ("random", False, "unicost"): lambda: None,
+            ("random", True, "unicost"): lambda: None,
+            ("greedy", False, "unicost"): lambda: None,
+            ("greedy", True, "unicost"): lambda: None,
+            ("exact", True, "unicost"): lambda: None,
+            ("exact", False, "unicost"): lambda: None
+        }
+
+        # If the entry of in the dictionary matches below argument set, then corresponding lambda function is retrieved
+        options = option_map.get((optimization_method, selection_size is not None, cost_metric))
+        if options:
+            options()
+        else:
+            feature_column = self.featurization_method.fit_transform(input_df)
+            self.features = np.array([eval(txt_feature) if isinstance(txt_feature, str)
+                                      else txt_feature for txt_feature in feature_column.tolist()])
+            if len(self.features) != self._num_cols:
```

Comment:
can we please turn these a if-raiseError statements into checkTrue() 

We should always use checkTrue() to raise these asserts.

Reply:
(don't forgot to negate the conditions as required when replacing with checkTrue)

Reply:
Done! I used check_true(true condition, ValueError())